File: 0404-sum-of-left-leaves/0404-sum-of-left-leaves.js
const sumOfLeftLeaves = (root) => {
  let sum = 0;
  const search = (n, isLeft) => {
    if (n.left) search(n.left, true)
    if (n.right) search(n.right)
    if (!n.left && !n.right && isLeft) sum += n.val;
  } 
  search(root);
  return sum;
};

File: 0404-sum-of-left-leaves/0404-sum-of-left-leaves.py
# class Solution:
#   def sumOfLeftLeaves(self, root: Optional[TreeNode]) -> int:
#     sum = 0
#     queue = [root]
#     while queue:
#       node = queue.pop()
#       if node.left:
#         if not node.left.left and not node.left.right:
#           sum += node.left.val
#         queue.append(node.left)
#       if node.right:
#         queue.append(node.right)
#     return sum

class Solution:
  def sumOfLeftLeaves(self, node: Optional[TreeNode]) -> int:
    sum = 0
    if node.left:
      if not node.left.left and not node.left.right:
        sum += node.left.val
      sum += self.sumOfLeftLeaves(node.left)
    if node.right:
      sum += self.sumOfLeftLeaves(node.right)
    return sum
        